torchrl/data/replay_buffers/samplers.py

```diff
@@ -1076,9 +1076,24 @@ def _tensor_slices_from_startend(self, seq_length, start, storage_length):
         # seq_length is a 1d tensor indicating the desired length of each sequence
 
         if isinstance(seq_length, int):
```

Comment:
I also looked into the possibility of speeding up the case where `seq_length` is a tensor. It seems a lot less straightforward, and I'm not entirely sure if we can get a speedup comparable to the `int` case. Since the sequence lengths are all different, it inherently requires doing something that is equivalent to calling `torch.arange` multiple times and `torch.cat`ing the results together.

I can continue to investigate if you'd like--I just didn't want to invest too much time in it without discussing it first